sql/expression/function/json_object.go

```diff
@@ -96,10 +96,11 @@ func (j JSONObject) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 			return nil, err
 		}
 		if i%2 == 0 {
-			var ok bool
-			if key, ok = val.(string); !ok {
-				return nil, sql.ErrInvalidType.New(expr.Type())
+			val, err2 := sql.LongText.Convert(val)
```

Comment:
More idiomatic to just re-use the err var here instead of introducing a new one. A second `err` will shadow the one from the outer scope, which is what you want.

Reply:
Ahh, thanks. I didn't realize I could use := with an existing, defined var name. Fixed now.